src/main.rs

```diff
@@ -19,12 +23,17 @@ use parking_lot::ReentrantMutex;
 use std::{
 	error::Error,
 	io::stdout,
+	sync::{
+		Arc,
+		Mutex,
+	},
 };
 use tracing_subscriber::fmt::writer::MakeWriterExt;
 
 lazy_static! {
 	static ref LUA: ReentrantMutex<mlua::Lua> = ReentrantMutex::new(mlua::Lua::new());
 	static ref CONFIG: Config = Config::default();
+	static ref STRATA_STATE: Arc<Mutex<GlobalState<dyn Backend>>> = Arc::new(Mutex::new(<GlobalState::new()));
```

Comment:
```suggestion
	static ref STRATA_STATE: ReentrantMutex<GlobalState<dyn Backend>> = ReentrantMutext::new(GlobalState::new());
```
You don't need the `Arc` since ownership isn't shared but is static instead. I also think it's better to have a reentrent mutex to avoid mistakes leading to deadlocks, since the state will be used a lot. Maybe we will want an `RwLock` instead? Not sure, but it's easy to change later.

Note: I see where you were going with the `Backend` generic here, but since it's *runtime* genericity you end up using `dyn` and the whole generic becomes pointless. You could drop it entirely, and use `Box<dyn Backend>` in `StrataState` directly. Or maybe just use an enum? It depends on what the backend contains and does exactly.

Reply:
Okay, I'll make a new commit soon.